src/pages/Timeline.js

```diff
@@ -1,19 +1,112 @@
 import styled from "styled-components";
+import React from "react";
 
 import Header from "../components/Header";
 import PublicationForm  from "../components/PublicationForm.js";
+import { getAllPostRequest } from "../services/api";
+import ApplicationContext from "../contexts/ApplicationContext.js";
+import Publication from "../components/Publication.js";
+import Trending from "../components/Trending";
 
 export default function Timeline() {
+    const [posts, setPosts] = React.useState(null);
+    const { userToken } = React.useContext(ApplicationContext);
+    
+    const config = {
+        headers: {
+          Authorization: `Bearer ${userToken}`,
+        }
+    };
+    React.useEffect(() => {
+        async function data(){
+            const response = await getAllPostRequest(config);
+            if(response.status === 200){
+                console.log(response.data);
+                setPosts([...response.data]);
+            }else{
+                alert("An error occured while trying to fetch the posts, please refresh the page")
+            };
+        };
+        data()
+    },[]);
+    
+    function checkForPosts (){
+        if(posts === null){
+            return(
+                <TextContainer>
+                    <h2>Loading...</h2>
+                </TextContainer>
+            );
+        }else if(posts.length === 0){
+            return (
+                <TextContainer>
+                    <h2>There are no posts yet</h2>
+                </TextContainer>
+            );
+        }else{
+            return(
+                posts.map((item,index)=>(
+                    <Publication  
+                        key={index}
```

Comment:
Please change this to item.id because it will be needed for deleting and editing.

Reply:
Done.